test/runner/e2e_update_rel_test.cpp

```diff
@@ -0,0 +1,165 @@
+        vector<string>{"", "21", "22", "23", "24", "210", "26", "27", "28", "29", "300"};
+    sortAndCheckTestResults(actualResult, expectedResult);
+}
+
+TEST_F(UpdateRelTest, UpdateStrProp) {
+    ASSERT_TRUE(conn->query(getUpdateRelQuery("animal" /* srcTable */, "person" /* dstTable */,
+                                "knows" /* relation */, 15 /* srcID */, 0 /* dstID */,
+                                "set e.place = 'waterloo'"))
+                    ->isSuccess());
+    ASSERT_TRUE(conn->query(getUpdateRelQuery("animal" /* srcTable */, "person" /* dstTable */,
+                                "knows" /* relation */, 16 /* srcID */, 0 /* dstID */,
+                                "set e.place = 'kitchener'"))
+                    ->isSuccess());
+    ASSERT_TRUE(conn->query(getUpdateRelQuery("animal" /* srcTable */, "person" /* dstTable */,
+                                "knows" /* relation */, 19 /* srcID */, 0 /* dstID */,
+                                "set e.place = null"))
+                    ->isSuccess());
+    auto result = conn->query(
+        "MATCH (a:animal)-[e:knows]->(:person) where a.ID >= 10 and a.ID <= 20 return e.place");
+    auto actualResult = TestHelper::convertResultToString(*result);
+    auto expectedResult = vector<string>{"990990990", "989", "988988988", "987", "986986986",
+        "waterloo", "kitchener", "983", "982982982", "", "980980980"};
+    sortAndCheckTestResults(actualResult, expectedResult);
+}
+
+TEST_F(UpdateRelTest, UpdateListProp) {
+    ASSERT_TRUE(conn->query(getUpdateRelQuery("animal" /* srcTable */, "person" /* dstTable */,
+                                "knows" /* relation */, 43 /* srcID */, 0 /* dstID */,
+                                "set e.tag = [\"updated property1\", \"50\"]"))
+                    ->isSuccess());
+    ASSERT_TRUE(
+        conn->query(getUpdateRelQuery("animal" /* srcTable */, "person" /* dstTable */,
+                        "knows" /* relation */, 47 /* srcID */, 0 /* dstID */, "set e.tag = null"))
+            ->isSuccess());
+    ASSERT_TRUE(conn->query(getUpdateRelQuery("animal" /* srcTable */, "person" /* dstTable */,
+                                "knows" /* relation */, 49 /* srcID */, 0 /* dstID */,
+                                "set e.tag = [\"updated property3\", \"54\"]"))
+                    ->isSuccess());
+    auto result = conn->query(
+        "MATCH (a:animal)-[e:knows]->(:person) where a.ID >= 40 and a.ID <= 50 return e.tag");
+    auto actualResult = TestHelper::convertResultToString(*result);
+    auto expectedResult = vector<string>{"[960960960]", "[959]", "[958958958]",
+        "[updated property1,50]", "[956956956]", "[955]", "[954954954]", "", "[952952952]",
+        "[updated property3,54]", "[950950950]"};
+    sortAndCheckTestResults(actualResult, expectedResult);
+}
+
+TEST_F(UpdateRelTest, UpdateLargeList) {
+    ASSERT_TRUE(conn->query(getUpdateRelQuery("person" /* srcTable */, "person" /* dstTable */,
```

Comment:
Do you make sure that some of these updates will not happen to the first page of the large list? I'm not sure what this test tests given that you have a more comprehensive test (UpdateEachElemOfLargeList) below. Maybe you should remove this test.

Reply:
I think i can remove this test.